engine/components-api/src/main/java/com/cloud/resource/ResourceManager.java

```diff
@@ -61,6 +61,14 @@ public interface ResourceManager extends ResourceService, Configurable {
                     + "To force-stop VMs, choose 'ForceStop' strategy",
             true, ConfigKey.Scope.Global, null, null, null, null, null, ConfigKey.Kind.Select, "Error,Migration,ForceStop");
 
+    ConfigKey<String> SystemVmPreferredArchitecture = new ConfigKey<>("Advanced"
+            , String.class
+            , "system.vm.preferred.architecture"
+            , ""
+            , "Preferred architecture for the system VMs including virtual routers"
+            , true
+            , ConfigKey.Scope.Zone);
```

Comment:
Can we have the commas at the end of each line instead of at the begginning?

Reply:
Is it possible to use this [constructor](https://github.com/apache/cloudstack/blob/4.20/framework/config/src/main/java/org/apache/cloudstack/framework/config/ConfigKey.java#L160-L176) to define the configKey with kind: Select and specify options which probably would be helpful in the UI.

Reply:
Done